components/gitpod-db/src/typeorm/app-installation-db-impl.ts

```diff
@@ -49,14 +49,15 @@ export class TypeORMAppInstallationDBImpl implements AppInstallationDB {
         installationID: string,
     ): Promise<AppInstallation | undefined> {
         const repo = await this.getRepo();
-        const qb = repo
+        const installation = await repo
             .createQueryBuilder("installation")
             .where("installation.installationID = :installationID", { installationID })
-            .andWhere('installation.state != "uninstalled"')
-            .orderBy("installation.lastUpdateTime", "DESC")
-            .limit(1);
+            .andWhere("installation.platform = :platform", { platform })
+            .andWhere("installation.state IN ('claimed.user', 'claimed.platform', 'installed')")
+            .orderBy("installation.creationTime", "DESC")
```

Comment:
Turns out creationTime and lastUpdateTime are virtually identical: `creationTime timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),  lastUpdateTime timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6)` :see_no_evil: